src/nixos/machines/tupac/config/nvidia.nix

```diff
@@ -0,0 +1,34 @@
+{ config, lib, ... }:
+
+# Nvidia management of TUPAC including optimus
+
+let
+	inherit (lib) mkIf;
+in {
+	hardware.nvidia = {
+		modesetting.enable = true; # Enable modesetting, which is needed for Wayland compositors
+
+		powerManagement.finegrained = true; # Turns off GPU when not in use
```

Comment:
Experimental and only works on modern Nvidia GPUs (Turing or newer). It's got issues and is buggy sometimes.

Reply:
> Experimental and only works on modern Nvidia GPUs (Turing or newer). It's got issues and is buggy sometimes. -- @TanvirOnGH (https://github.com/NiXium-org/NiXium/pull/74#discussion_r1660035331)

Context to the system: Tupac is peak high-end portable workstation with fully open-source firmware (pending deployment) for our environment that has binned RTX4060M as RTX4070M, RTX4080M and RTX4090M are too energy inefficient and can't be cooled in a reasonable formfactor (plans to upgrade on RTX4090M which would be downclocked significantly and only allowed to go full power when connected to water cooler ala XMG Oasis, but it's still too energy inefficient to need a custom BIOS) .

It's supported on the GPU and was tested to confirm that it works reliably, but i am monitoring it just to be sure, without it the battery life seems to suffer by ~2 h.